clients/drcachesim/tests/allasm-repstr-basic-counts.templatex

```diff
@@ -29,7 +29,7 @@ Total counts:
            0 total physical address \+ virtual address marker pairs
            0 total physical address unavailable markers
            4 total other markers
-           0 total encodings
+         102 total encodings
```

Comment:
102 encodings for 26 unique instrs? Where are the "extra" ones coming from?

Reply:
Currently it is emitting encodings for every dynamic instance.  In fact it looks like I did not even put a TODO about omitting for repeats in each thread.  Actually I didn't even think about *how* to do that: hmm.  I don't think it's trivial.  It would have to involve per-thread tables and fragment_delete events -- but if a thread-shared fragment is deleted how do we remove from every thread's table?  I think at this point since online analyzers looking at encodings is not a priority, the best course of action is to put some TODOs in here and maybe leave the issue open and put in this PR as is w/ the extra overhead.  Agreed?

Reply:
A possible alternative would be to put in a table in each thread and assume no code deletion/changes with TODOs about that.  That would take a little work but it is straightforward; it would reduce overhead at the cost of accuracy.  Online is already slow-ish though and this is an off-by-default option so I would still vote for not doing anything now.

Reply:
> Currently it is emitting encodings for every dynamic instance. 

Oh right. But even then it's 98 vs 102; there seem to be 4 extra ones.

> omitting for repeats in each thread.

If we're outputting for each dynamic instance, then it's fine to repeat for each thread. I vote to keep online simple for now, with an XXX comment noting the possible optimizations.

Reply:
> But even then it's 98 vs 102; there seem to be 4 extra ones.

This is counting total records so my guess is that 4 of them are long and need 2 records each.  I will check to confirm.

Reply:
OK the other 4 are the non-fetches instrs: 98 fetched and 4 non-fetched (the rep loops).

I added a big comment about the repeated encodings and updated the issue.